crates/runtime/src/lib.rs

```diff
@@ -645,17 +653,48 @@ impl Runtime {
             }
         });
 
-        // Wait for all tasks to complete
-        let load_result = tokio::try_join!(
+        let components = vec![
             task_history,
             results_cache,
             datasets,
             catalogs,
-            models_and_evals
-        );
+            models_and_evals,
+        ];
+
+        // Signal that the load must be canceled if the runtime is shut down before the components are loaded
+        let cancel_loading = CancellationToken::new();
 
-        if let Err(err) = load_result {
-            tracing::error!("Could not start the Spice runtime: {err}");
+        // Wait for all components to load returning the first error
+        // or canceling spawned tokio tasks if the runtime is shutting down
+        let load_result = self
+            .start_server_component(
```

Comment:
I think calling this a `server_component` is no longer accurate. Perhaps a better name is `runtime_task`?

Reply:
Updated: https://github.com/spiceai/spiceai/pull/5306/commits/dc414742d05292b4990cfeb88ac757f80e382fd0